src/stage_node.cpp

```diff
@@ -172,9 +176,71 @@ int StageNode::callback_init_stage_model(Stg::Model * mod, StageNode * node)
       }
     }
   }
+
+  if (dynamic_cast<Stg::Model *>(mod)) {
+    Stg::Model * model = dynamic_cast<Stg::Model *>(mod);
+    std::string model_name(mod->TokenStr().c_str());
+    if(model_name.find("obstacle") != std::string::npos){ // such that only the intended models become "objects"
+      RCLCPP_INFO(node->get_logger(), "New Object \"%s\"", model_name.c_str());
+      auto object = std::make_shared<Object>(
+        node->objects_.size(),
+        model->GetGlobalPose(), mod->TokenStr(), node);
+      node->objects_.push_back(object);
+      object->model = model;
+    }
+  }
+
   return 0;
 }
 
+void StageNode::publish_object_visualization(StageNode * node)
+{
+  rclcpp::QoS qos(10);
+  rclcpp::Publisher<visualization_msgs::msg::MarkerArray>::SharedPtr pub_object = node->create_publisher<visualization_msgs::msg::MarkerArray>("obstacle_viszualization", qos);
```

Comment:
I assume this the reason why we do not see the topic in rviz2: recreating the temporary publisher seems a bit of an overhead. Please set it instead as a member variable of the StageNode class (https://github.com/ubica-robotics/stage_ros2/pull/3/files#diff-7293bd38fd78d68fd4dc10ec64926f03dff010cb13dc688979189b15160ad205R212) and only publish in this function with it as you already do below.